### PR TITLE
Initial MOOSE command bash autocomplete work

### PR DESCRIPTION
**Pull request by dschwen**
Check this out and give feedback please. This has been a long time itch for me!

Refs #84

### CODE REVIEW

**Comment by milljm** [comment]
If you do not want to make it default for everyone, you can adjust that in the package creation process here:

https://github.com/idaholab/package_builder/blob/devel/create_redistributable/pkg/OSX/environment_post.sh

This is where the actual writing to the users's ~/.bash_profile takes place (for Darwin machines). Just to give you an idea. The linux installers don't ever modify a users profile. Instead, we print a quick "Paste the following into your bash profile" helper message.

However, if you believe this should be enabled all the time by everyone... no need to change anything. I still don't know exactly what it is this does yet :)

**Comment by dschwen** [comment]
I think it should be default. It adds bash autocomplete support (tab) for valid options for `./run_tests` and `./moosedocs.py`.

File: build_from_source/template/moose_profile [hunk 1]
```diff
@@ -54,6 +54,9 @@ else
   export MOOSE_JOBS=\`cat /proc/cpuinfo | grep processor | wc -l\`
   module load moose-dev-gcc moose-tools
 fi
+
+# Load custom MOOSE command bash autocomplete rules
+source $PACKAGES_DIR/environments/autocomplete_moose
```
Comment (milljm) [inline build_from_source/template/moose_profile]: Should we make this an optional feature?

```bash
# If, anyone wants to use autocomplete:
if [ -n "\$MOOSE_AUTOCOMPLETE" ]; then
  source $PACKAGES_DIR/environments/autocomplete_moose
fi
```

**Review by milljm — APPROVED** [review]
Ok then, sounds like an awesome addition

**Comment by dschwen** [comment]
Don't merge yet, I think I'll add to this PR some more.

**Comment by dschwen** [comment]
Ok, this is outdated now anyways that @aeslaughter changed the `./moosedocs.py` command line options. I'll update this.

**Comment by aeslaughter** [comment]
We need a way to generate these lists and include them in MOOSE, because I know that options will come and go from moosedocs.py and this will get out-of-date constantly. There must be an argparse to bash autocomplete tool already.

**Comment by aeslaughter** [comment]
https://argcomplete.readthedocs.io/en/latest/

**Comment by dschwen** [comment]
neato!

**Comment by brianmoose** [comment]
@milljm luckily noticed that this broke the package (well, sourcing moose_profile).
You need to escape the `$` in most places, otherwise it is processed as template variables.
See `build_from_source/template/moose_profile` for examples.

**Pull request merged**